lib/prefix.c

```diff
@@ -556,6 +556,31 @@ int prefix_match(const struct prefix *n, const struct prefix *p)
 	return 1;
 }
 
+/* If n includes p prefix then return 1 else return 0. */
+int prefix_opaque_match(const struct prefix *n, const struct prefix *p)
+{
+	int offset;
+	const u_char *np, *pp;
+
+	if (n->family != AF_FLOWSPEC)
+		return 0;
+
+	/* prefixlen is unused. look at fs prefix len */
+	if (n->u.prefix_flowspec.prefixlen > p->u.prefix_flowspec.prefixlen)
+		return 0;
+
+	/* Set both prefix's head pointer. */
+	np = (const u_char *)&n->u.prefix_flowspec.ptr;
+	pp = (const u_char *)&p->u.prefix_flowspec.ptr;
+
+	offset = n->u.prefix_flowspec.prefixlen;
```

Comment:
So we know from line 569 n->u.prefix_flowspec.prefixlen <= p->u.prefix_flowspec.prefixlen.  Now suppose n's prefix len is 10 while p's prefixlen is 20, if I am reading the while loop below correctly we only look at the first 10 bytes of n and p and if they are the same we stop looking.  Why are we not looking at bytes 11-20?

Reply:
this is probably ok.  But I really don't understand what a flowspec prefix is yet.

Reply:
from RFC5575:
This NLRI is treated as an opaque bit string prefix by
  BGP.  Each bit string identifies a key to a database entry with which
  a set of attributes can be associated.

about your remark, there is no meaning of comparing two prefixes if the length is different.
If length is different, prefixes are different. that is all.
I made the fix.